### ISSUE TITLE
Web rtc from cloudflare

### ISSUE THREAD

**Issue by torik988** [balzack/databag#99]
Hello, I've been using Databag for almost 2 month now, and I've really enjoyed your project. However, from the beginning, I wanted to use WebRTC calls but didn't have a public IP to host the server.S i decided to use cloudflare tunnels to allons access to my server. Recently, I discovered that Cloudflare offers a free call WebRTC server service. On the main dashboard of your account, you can choose to call and create a token to access their service. However, the authentication token changes frequently, and they ask the user to create their backend server to generate the username and password. I wanted to know if it was possible to add a place in the Databag admin dashboard where you could add your token and app ID, and Databag would automatically generate the username and password and use it. If I'm unclear, I tried my best (I'm French)😅. If you have any questions, I'm still open. Have a nice day. TORIK

**Comment by balzack** [comment]
I think I found what you are referring to:
https://developers.cloudflare.com/calls/turn/
It seems straightforward to integrate.

I am finishing a release to support MFA (TOTP) in the next week or two, and then I will integrate the cloudflare webrtc shortly after. (assuming it is as easy as it seems)

Thank you for using this project! I’m really happy to hear you like it. I’m also French btw :)

**Comment by balzack** [comment]
Cloudflare setup options have been added to the admin dashboard in the latest release. Can confirm it works for you when you get a chance? Thanks for using this project!

**Comment by torik988** [comment]
hey i've sorry for the long responce, i did tried tu use cloudflare webrtc but i've got an error. 
![image](https://github.com/user-attachments/assets/b407690b-c5b2-404a-ae45-2615acbcb21f)

**Comment by torik988** [comment]
i just tried it and it seems to work, tanks :)

**Comment by balzack** [comment]
It should use the webrtc settings of the person calling. In your test, was the person on the other server initiating the call? Are both servers running the latest version?

**Comment by torik988** [comment]
j'ai deux comptes: 
le compte A sur mon serveur personnel avec le service turn de cloudflare pour les appels webrtc,
le compte B sur le serveur (databag.coredb.org).
lors de mon test, mon serveur personnel étais sous la version 0.1.15 (latest).
l'appel a été démarré depuis le serveur B

**Comment by balzack** [comment]
Je viens de faire l'appel depuis databag.coredb.org et depuis mon serveur et ça marche dans les deux sens. (mon serveur et configuré avec cloudflare) Pour le moment je ne vois pas pourquoi ça ne marche pas pour vous, mais je vais continue à recherche.

**Comment by torik988** [comment]
Re je viens de retester avec une autre personne en apel vocal sur le meme serveur et il semble que lorsque la camera est activée, cela semble couper le micro et apres, il faut relancer un nouvel appel pour le retrouver. pour le test entre les serveurs, je vais reesayer et vous redis

**Comment by torik988** [comment]
dans mon cas, en ,initiant l'appel du serveur coredb vers mon serveur, lorsque je lance l'apppel, jai bien le message cote receveur, mais lorsque je rejoins, l'appel se coupe, la fenetre d'appel se ferme, et le compte de serveur A se marque comme Hors Sync

**Comment by torik988** [comment]
effectivement, après un autre test, lorsque l'on effectue les appels sur l'interface web tout fonctionne, même entre plusieurs serveurs :)
EDIT
apres avoir essayé sur l'application mobile, l'appel fonctionne bien entre comptes d'un meme serveur, mais dés lors que les comptes sont sur deux serveurs, ça ne fonctionne plus sur mobile.

**Comment by torik988** [comment]
J'ai décide de refaire un message plus claire qui résume mes découvertes et lister les deux problèmes rencontrés:

- Problème de Caméra et Micro :
   Lors d'un appel vocal sur le même serveur, l'activation de la caméra semble couper le micro.
   Pour rétablir le micro, il faut relancer un nouvel appel.

- Problème d'appel entre Différents Serveurs :
  Lors d'un appel initié du serveur "coredb" vers "mon serveur" ou l'inverse:
  Le récepteur reçoit bien la notification d'appel sur l'application et sur l' ui web
  Les appels sont fonctionnels, SAUF _**lorsque l'un des clients est sur l'appli mobile**_. il n'y a pas ce problème lors d'un appel sur le même serveur.

  Details:
  Lorsque le récepteur rejoint l'appel depuis l'application mobile, l'appel se coupe, la fenêtre d'appel se ferme, et le compte du  
  du serveur "coredb" est marqué comme "Hors Sync". c'est la même chose lorsque l'appel est initie depuis l'appli mobile sur mon 
  serveur vers "coredb"

### En résumé:

- Fonctionnement sur l'Interface Web :
  Lors des tests sur l'interface web, tout fonctionne bien, même entre plusieurs serveurs.

- Fonctionnement sur l'Application Mobile :
  Les appels fonctionnent entre comptes d'un même serveur.
  Les appels ne fonctionnent pas lorsque les comptes sont sur deux serveurs différents.

J'espère être plus clair :)

**Comment by balzack** [comment]
C'est clair ce qui ne marche pas pour vous, mais j'essaye de faire les mêmes tests, et ça continue à marcher. Est-ce que votre application mobil est mise à jour, et est-ce que vous utilise iphone ou Android? Peut-être il y-a quelque chose avec le réseau mobil? Est-ce que vous pouvez essayer avec mobil sur Wi-Fi? Je suis désolé que ça ne marche pas comme prévu pour vous pour l'instant.

**Comment by torik988** [comment]
Application mobile sous android 13 et sous la derniere version (dl aujourdhui)

**Comment by itsdoublearon** [comment]
The quality of cloudflare is great, only problem I'm running into is calls don't hangup for the other person when I hang up, and vice versa. Can this be looked into please.

**Comment by balzack** [comment]
Does it consistently not hang up for the other person? Is this on browser only or mobile also? As long as I can reproduce, it should be easy to fix. Thanks for reporting!

**Comment by itsdoublearon** [comment]
It's both browser and mobile app. I'm on the latest version of the app on both ios and server side.
